lib/ember-cli/path_set.rb

```diff
@@ -0,0 +1,96 @@
+module EmberCLI
+  class PathSet
+    def self.define_path(name, &definition)
+      define_method name do
+        ivar = "@_#{name}_path"
+        if instance_variable_defined?(ivar)
+          instance_variable_get(ivar)
+        else
+          instance_exec(&definition).tap{ |value| instance_variable_set ivar, value }
+        end
+      end
+    end
+
+    def initialize(app)
+      @app = app
+    end
+
+    define_path :root do
+      path = app_options.fetch(:path){ default_root }
+      pathname = Pathname.new(path)
+      pathname.absolute?? pathname : Rails.root.join(path)
```

Comment:
@seanpdoyle in this case `Rails.root.join(path)` and `Rails.root.join(pathname)` will do exactly the same. `path` is shorter :)